src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs

```diff
@@ -0,0 +1,708 @@
+            PerformNullabilityTest<uint?>();
+            PerformNullabilityTest<ulong?>();
+            PerformNullabilityTest<float?>();
+            PerformNullabilityTest<decimal?>();
+        }
+
+        private static void PerformNullabilityTest<T>()
+        {
+            string nullAsJson = "null";
+            string nullAsQuotedJson = $@"""{nullAsJson}""";
+
+            if (Nullable.GetUnderlyingType(typeof(T)) != null)
+            {
+                Assert.Null(JsonSerializer.Deserialize<T>(nullAsJson, s_optionReadAndWriteFromStr));
+                Assert.Equal(nullAsJson, JsonSerializer.Serialize(default(T)));
+            }
+            else
+            {
+                Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<T>(nullAsJson, s_optionReadAndWriteFromStr));
+                Assert.Equal("0", JsonSerializer.Serialize(default(T)));
+            }
+
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<T>(nullAsQuotedJson, s_optionReadAndWriteFromStr));
+        }
+
+        [Fact]
+        public static void ConvertersHaveNullChecks()
+        {
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Bytes[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.SBytes[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Shorts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Ints[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Longs[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.UShorts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.UInts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.ULongs[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Floats[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Doubles[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.Decimals[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableBytes[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableSBytes[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableShorts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableInts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableLongs[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableUShorts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableUInts[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableULongs[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableFloats[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableDoubles[10]);
+            RunConvertersHaveNullChecksTest(JsonNumberTestData.NullableDecimals[10]);
+        }
+
+        private static void RunConvertersHaveNullChecksTest<T>(T number)
```

Comment:
Can you please elaborate on what is this method testing?

Reply:
In the initial commit of this PR, there were null checks for the `JsonSerializerOptions` instance in the converter `Read`/`Write` methods. I've removed them now since the code has been restructured to include `JsonNumberHandlingAttribute` usage. This test has thus been removed.